docs/examples/good_practices/many_tasks_per_gpu/job.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+#SBATCH --gpus=1
+#SBATCH --ntasks=2
+#SBATCH --mem=16G
+#SBATCH --time=00:15:00
+
+
+# Echo time and hostname into log
+echo "Date:     $(date)"
+echo "Hostname: $(hostname)"
+
+
+# Ensure only anaconda/3 module loaded.
+module --quiet purge
+# This example uses Conda to manage package dependencies.
+# See https://docs.mila.quebec/Userguide.html#conda for more information.
+module load anaconda/3
+module load cuda/11.7
+
+# Creating the environment for the first time:
+# conda create -y -n pytorch python=3.9 pytorch torchvision torchaudio \
+#     pytorch-cuda=11.7 -c pytorch -c nvidia
+# Other conda packages:
+# conda install -y -n pytorch -c conda-forge rich tqdm
+
+# Activate pre-existing environment.
+conda activate pytorch
+
+
+# Stage dataset into $SLURM_TMPDIR
+mkdir -p $SLURM_TMPDIR/data
+cp /network/datasets/cifar10/cifar-10-python.tar.gz $SLURM_TMPDIR/data/
+# General-purpose alternatives combining copy and unpack:
+#     unzip   /network/datasets/some/file.zip -d $SLURM_TMPDIR/data/
+#     tar -xf /network/datasets/some/file.tar -C $SLURM_TMPDIR/data/
+
+
+# Fixes issues with MIG-ed GPUs with versions of PyTorch < 2.0
+unset CUDA_VISIBLE_DEVICES
+
+# Execute Python script twice using srun to dispatch runs into tasks
+# As sbatch is configured to run 2 tasks, we allocate 1 task per run here using `--ntasks=1`
```

Comment:
Not quite! You should do a single srun, which will launch two tasks with the same command (because --ntasks=2 in this context, not passing it uses 2 tasks)

Reply:
Ok ! Fait! https://mila-docs--228.org.readthedocs.build/en/228/examples/good_practices/many_tasks_per_gpu/index.html